### PR TITLE
chore: move some lemmas from Analysis.Normed.Group.Basic to Lemmas

### PR DESCRIPTION
**Pull request by Ruben-VandeVelde**
This reduces the longest pole a bit: it now goes from Mathlib.Analysis.Normed.Field.Basic through Mathlib.Analysis.Normed.Group.Lemmas and Mathlib.Analysis.Normed.Group.Basic rather than through Mathlib.Analysis.Normed.Group.Basic and then Mathlib.Topology.MetricSpace.{IsometricSMul, Isometry, Antilipschitz}.

Files that import Normed.Group.Lemmas rather than Normed.Group.Basic do gain some dependencies through Topology.Instances.NNReal, though.

---

- [x] depends on: #13894
- [x] depends on: #13895
- [x] depends on:  #13945 

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by Ruben-VandeVelde** [comment]
!bench

**Comment by Ruben-VandeVelde** [comment]
Scratch that, it does help some with the longest pole:
```diff
-Analyzing Mathlib at 913dc471aee5ed0e647c470ada11517708044991
+Analyzing Mathlib at 205bf63649c144dff7f61f909008bc1c9f39dc7c
 file                                                           | instructions | (cumulative) | parallelism
 -------------------------------------------------------------- | ------------ | ------------ | -----------
-Mathlib.Analysis.Normed.Field.Basic                            |        88555 |      1783135 | x7.65
-Mathlib.Analysis.Normed.Group.Basic                            |       101596 |      1694579 | x6.73
-Mathlib.Topology.MetricSpace.IsometricSMul                     |        30947 |      1592983 | x6.54
-Mathlib.Topology.MetricSpace.Isometry                          |        13629 |      1562035 | x5.40
-Mathlib.Topology.MetricSpace.Antilipschitz                     |         8221 |      1548406 | x5.44
+Mathlib.Analysis.Normed.Field.Basic                            |        88554 |      1733842 | x7.87
+Mathlib.Analysis.Normed.Group.Lemmas                           |        35627 |      1645287 | x7.03
+Mathlib.Analysis.Normed.Group.Basic                            |        69477 |      1609660 | x5.28
 Mathlib.Topology.MetricSpace.Bounded                           |        13968 |      154018_ | x5.45
```

**Comment by YaelDillies** [comment]
Okay so now we need `Analysis.Normed.Field.Basic` to not import `Analysis.Normed.Group.Lemmas`. Can you try doing that?

**Comment by Ruben-VandeVelde** [comment]
!bench

**Comment by Ruben-VandeVelde** [comment]
This now flips the pole back to ring theory / linear algebra:
```diff
-Analyzing Mathlib at 19f8cc69f95fdcfea6db28055016c41be28791ed
+Analyzing Mathlib at 99a86326b653d4d87216599398f615ff107cdfbd
 | file                                                           | instructions | cumulative | parallelism |
 | :------------------------------------------------------------- | -----------: | ---------: | :---------: |
-| Mathlib                                                        |         7935 |    5337803 |    22.91    |
-| Mathlib.NumberTheory.Cyclotomic.Three                          |        33316 |    5329868 |    9.67     |
-| Mathlib.NumberTheory.Cyclotomic.Rat                            |       116595 |    5296551 |    9.66     |
-| Mathlib.NumberTheory.Cyclotomic.Discriminant                   |        42831 |    5179956 |    9.85     |
-| Mathlib.NumberTheory.NumberField.Discriminant                  |       215914 |    5137124 |    9.77     |
-| Mathlib.NumberTheory.NumberField.CanonicalEmbedding.ConvexBody |       187383 |    4921210 |    10.14    |
-| Mathlib.MeasureTheory.Measure.Lebesgue.VolumeOfBalls           |        63349 |    4733827 |    8.24     |
-| Mathlib.Analysis.SpecialFunctions.Gamma.BohrMollerup           |        37714 |    4670478 |    8.31     |
-| Mathlib.Analysis.SpecialFunctions.Gaussian.GaussianIntegral    |        38268 |    4632763 |    8.37     |
-| Mathlib.Analysis.SpecialFunctions.Gamma.Basic                  |        39764 |    4594495 |    8.42     |
-| Mathlib.Analysis.MellinTransform                               |        40611 |    4554730 |    8.48     |
-| Mathlib.Analysis.SpecialFunctions.ImproperIntegrals            |        16161 |    4514119 |    8.53     |
-| Mathlib.Analysis.SpecialFunctions.JapaneseBracket              |        45350 |    4497958 |    8.29     |
-| Mathlib.Analysis.SpecialFunctions.Integrals                    |       115989 |    4452607 |    8.32     |
-| Mathlib.Analysis.SpecialFunctions.Pow.Deriv                    |       121928 |    4336618 |    7.03     |
-| Mathlib.Analysis.SpecialFunctions.Trigonometric.Deriv          |       115467 |    4214690 |    6.98     |
-| Mathlib.Analysis.SpecialFunctions.ExpDeriv                     |        32381 |    4099223 |    7.13     |
-| Mathlib.Analysis.Complex.RealDeriv                             |        47189 |    4066842 |    7.01     |
-| Mathlib.Analysis.Calculus.ContDiff.Basic                       |       439809 |    4019653 |    5.48     |
-| Mathlib.Analysis.Calculus.FDeriv.Mul                           |       437357 |    3579843 |    5.74     |
-| Mathlib.Analysis.Calculus.FDeriv.Bilinear                      |        38129 |    3142486 |    6.40     |
-| Mathlib.Analysis.Calculus.FDeriv.Prod                          |       101273 |    3104356 |    6.47     |
-| Mathlib.Analysis.Calculus.FDeriv.Linear                        |        15992 |    3003083 |    6.64     |
-| Mathlib.Analysis.NormedSpace.BoundedLinearMaps                 |        95983 |    2987090 |    6.62     |
-| Mathlib.Analysis.NormedSpace.Multilinear.Basic                 |       351668 |    2891107 |    6.56     |
-| Mathlib.Analysis.NormedSpace.OperatorNorm.NormedSpace          |        75121 |    2539438 |    7.20     |
-| Mathlib.Analysis.NormedSpace.OperatorNorm.Bilinear             |       160737 |    2464317 |    7.37     |
-| Mathlib.Analysis.NormedSpace.OperatorNorm.Basic                |       106411 |    2303579 |    7.82     |
-| Mathlib.Analysis.LocallyConvex.WithSeminorms                   |       101244 |    2197168 |    8.02     |
-| Mathlib.Analysis.LocallyConvex.Bounded                         |        40219 |    2095923 |    7.44     |
-| Mathlib.Analysis.Seminorm                                      |       173244 |    2055704 |    7.36     |
-| Mathlib.Analysis.LocallyConvex.Basic                           |        38961 |    1882459 |    7.83     |
-| Mathlib.Analysis.NormedSpace.Basic                             |        29862 |    1843497 |    7.55     |
-| Mathlib.Analysis.Normed.MulAction                              |        18394 |    1813635 |    7.64     |
-| Mathlib.Analysis.Normed.Field.Basic                            |        88952 |    1795240 |    7.70     |
-| Mathlib.Analysis.Normed.Group.Uniform                          |        24154 |    1706288 |    6.74     |
-| Mathlib.Analysis.Normed.Group.Basic                            |        65677 |    1682133 |    6.75     |
-| Mathlib.Topology.Instances.Rat                                 |        12822 |    1616456 |    6.96     |
-| Mathlib.Topology.Instances.Real                                |        13931 |    1603633 |    7.00     |
-| Mathlib.Topology.Instances.Int                                 |         5136 |    1589701 |    5.42     |
-| Mathlib.Topology.Instances.Discrete                            |         6826 |    1584565 |    5.32     |
-| Mathlib.Topology.Metrizable.Uniformity                         |        13788 |    1577738 |    5.34     |
-| Mathlib.Topology.Metrizable.Basic                              |         4947 |    1563950 |    5.36     |
-| Mathlib.Topology.MetricSpace.Basic                             |         9752 |    1559002 |    5.38     |
-| Mathlib.Topology.MetricSpace.PseudoMetric                      |        66566 |    1549250 |    5.41     |
-| Mathlib.Topology.Order.DenselyOrdered                          |        86291 |    1482684 |    4.22     |
-| Mathlib.Topology.Order.IsLUB                                   |        15786 |    1396393 |    4.42     |
-| Mathlib.Topology.Order.LeftRightNhds                           |        33064 |    1380606 |    4.46     |
-| Mathlib.Topology.Order.Basic                                   |        36973 |    1347542 |    4.52     |
-| Mathlib.Topology.Order.OrderClosed                             |        30728 |    1310568 |    4.10     |
-| Mathlib.Topology.Separation                                    |        47492 |    1279840 |    4.18     |
-| Mathlib.Topology.Compactness.Lindelof                          |        14220 |    1232347 |    4.22     |
-| Mathlib.Topology.Compactness.SigmaCompact                      |         8248 |    1218127 |    4.25     |
-| Mathlib.Topology.Compactness.LocallyCompact                    |         5548 |    1209879 |    4.27     |
-| Mathlib.Topology.Compactness.Compact                           |        29101 |    1204330 |    4.29     |
-| Mathlib.Topology.Bases                                         |        32264 |    1175229 |    4.35     |
-| Mathlib.Topology.ContinuousOn                                  |        30410 |    1142964 |    4.44     |
-| Mathlib.Topology.Constructions                                 |        38285 |    1112554 |    4.54     |
-| Mathlib.Topology.Maps                                          |        12361 |    1074268 |    4.66     |
-| Mathlib.Topology.Order                                         |        17448 |    1061907 |    4.70     |
-| Mathlib.Topology.Defs.Induced                                  |         3550 |    1044458 |    4.76     |
-| Mathlib.Topology.Basic                                         |        31498 |    1040907 |    4.78     |
-| Mathlib.Topology.Defs.Filter                                   |         8102 |    1009409 |    4.88     |
-| Mathlib.Order.Filter.Ultrafilter                               |        12527 |    1001307 |    4.80     |
-| Mathlib.Order.Filter.Cofinite                                  |         6244 |     988779 |    4.79     |
-| Mathlib.Order.Filter.AtTopBot                                  |        65104 |     982535 |    4.80     |
-| Mathlib.Order.Filter.Bases                                     |        22197 |     917431 |    4.73     |
-| Mathlib.Order.Filter.Prod                                      |        16379 |     895233 |    4.45     |
-| Mathlib.Order.Filter.Basic                                     |        64178 |     878854 |    4.51     |
+| Mathlib                                                        |         7935 |    5307945 |    23.04    |
+| Mathlib.NumberTheory.Cyclotomic.Three                          |        33316 |    5300009 |    9.73     |
+| Mathlib.NumberTheory.Cyclotomic.Rat                            |       116589 |    5266693 |    9.71     |
+| Mathlib.NumberTheory.Cyclotomic.Discriminant                   |        42832 |    5150103 |    9.90     |
+| Mathlib.NumberTheory.NumberField.Discriminant                  |       215916 |    5107271 |    9.82     |
+| Mathlib.NumberTheory.NumberField.CanonicalEmbedding.ConvexBody |       187349 |    4891355 |    10.20    |
+| Mathlib.NumberTheory.NumberField.CanonicalEmbedding.Basic      |       147311 |    4704005 |    10.05    |
+| Mathlib.NumberTheory.NumberField.Embeddings                    |       132136 |    4556694 |    9.24     |
+| Mathlib.NumberTheory.NumberField.Norm                          |        48565 |    4424557 |    5.09     |
+| Mathlib.NumberTheory.NumberField.Basic                         |        40893 |    4375991 |    5.10     |
+| Mathlib.RingTheory.DedekindDomain.IntegralClosure              |        44385 |    4335098 |    5.14     |
+| Mathlib.RingTheory.Trace                                       |       119345 |    4290713 |    5.07     |
+| Mathlib.FieldTheory.Galois                                     |        90277 |    4171367 |    4.81     |
+| Mathlib.FieldTheory.PrimitiveElement                           |        64362 |    4081090 |    4.85     |
+| Mathlib.FieldTheory.IsAlgClosed.AlgebraicClosure               |        50979 |    4016727 |    4.91     |
+| Mathlib.FieldTheory.IsAlgClosed.Basic                          |       103606 |    3965747 |    4.87     |
+| Mathlib.FieldTheory.Normal                                     |       103704 |    3862141 |    4.95     |
+| Mathlib.FieldTheory.Extension                                  |        71099 |    3758436 |    5.05     |
+| Mathlib.FieldTheory.Adjoin                                     |       288813 |    3687336 |    5.12     |
+| Mathlib.FieldTheory.SplittingField.IsSplittingField            |        17525 |    3398523 |    5.46     |
+| Mathlib.RingTheory.Adjoin.Field                                |        24061 |    3380998 |    5.43     |
+| Mathlib.RingTheory.AdjoinRoot                                  |       130230 |    3356937 |    5.45     |
+| Mathlib.RingTheory.PowerBasis                                  |        46617 |    3226706 |    5.61     |
+| Mathlib.FieldTheory.Minpoly.Field                              |        33904 |    3180089 |    5.68     |
+| Mathlib.RingTheory.Algebraic                                   |        38813 |    3146184 |    5.73     |
+| Mathlib.RingTheory.IntegralClosure                             |       104771 |    3107371 |    5.78     |
+| Mathlib.LinearAlgebra.Matrix.Charpoly.LinearMap                |        40921 |    3002600 |    5.90     |
+| Mathlib.LinearAlgebra.Matrix.Charpoly.Coeff                    |        82790 |    2961678 |    5.97     |
+| Mathlib.LinearAlgebra.Matrix.Charpoly.Basic                    |        17770 |    2878887 |    6.05     |
+| Mathlib.RingTheory.PolynomialAlgebra                           |        37555 |    2861117 |    5.68     |
+| Mathlib.RingTheory.MatrixAlgebra                               |        31664 |    2823562 |    5.64     |
+| Mathlib.RingTheory.TensorProduct.Basic                         |       279826 |    2791897 |    5.69     |
+| Mathlib.LinearAlgebra.FiniteDimensional                        |       105512 |    2512071 |    6.22     |
+| Mathlib.LinearAlgebra.Dimension.DivisionRing                   |        45112 |    2406558 |    6.43     |
+| Mathlib.LinearAlgebra.Dimension.RankNullity                    |        28654 |    2361445 |    6.49     |
+| Mathlib.LinearAlgebra.Dimension.Constructions                  |        93278 |    2332791 |    6.53     |
+| Mathlib.Algebra.Module.Torsion                                 |        88107 |    2239512 |    6.70     |
+| Mathlib.GroupTheory.Torsion                                    |        18622 |    2151405 |    6.90     |
+| Mathlib.GroupTheory.PGroup                                     |        19508 |    2132782 |    6.95     |
+| Mathlib.GroupTheory.SpecificGroups.Cyclic                      |        43770 |    2113274 |    6.91     |
+| Mathlib.GroupTheory.Exponent                                   |        25385 |    2069503 |    6.30     |
+| Mathlib.Data.ZMod.Quotient                                     |        22196 |    2044118 |    6.27     |
+| Mathlib.RingTheory.Int.Basic                                   |         7322 |    2021922 |    6.07     |
+| Mathlib.RingTheory.PrincipalIdealDomain                        |        25904 |    2014599 |    6.08     |
+| Mathlib.RingTheory.UniqueFactorizationDomain                   |        84125 |    1988695 |    6.14     |
+| Mathlib.RingTheory.Noetherian                                  |       102349 |    1904570 |    6.33     |
+| Mathlib.RingTheory.OrzechProperty                              |         6886 |    1802221 |    6.38     |
+| Mathlib.RingTheory.Finiteness                                  |       102446 |    1795334 |    6.37     |
+| Mathlib.RingTheory.Ideal.Maps                                  |        50696 |    1692888 |    5.90     |
+| Mathlib.RingTheory.Ideal.Operations                            |        81589 |    1642191 |    6.05     |
+| Mathlib.Algebra.Algebra.Operations                             |        67260 |    1560601 |    6.29     |
+| Mathlib.Algebra.Algebra.Bilinear                               |        21605 |    1493340 |    5.41     |
+| Mathlib.LinearAlgebra.TensorProduct.Basic                      |       261076 |    1471735 |    4.85     |
+| Mathlib.Algebra.Module.Submodule.Bilinear                      |        14837 |    1210658 |    5.65     |
+| Mathlib.LinearAlgebra.Span                                     |        71703 |    1195821 |    5.62     |
+| Mathlib.LinearAlgebra.Basic                                    |        22109 |    1124118 |    5.62     |
+| Mathlib.Algebra.Module.Submodule.Range                         |        25886 |    1102008 |    5.44     |
+| Mathlib.Algebra.Module.Submodule.Ker                           |        26308 |    1076121 |    5.54     |
+| Mathlib.Algebra.Module.Submodule.Map                           |        41102 |    1049813 |    5.66     |
+| Mathlib.Algebra.Module.Submodule.Lattice                       |        31415 |    1008711 |    5.83     |
+| Mathlib.Algebra.Module.Submodule.Basic                         |        33567 |     977296 |    5.22     |
+| Mathlib.Algebra.Group.Submonoid.Membership                     |        30527 |     943728 |    4.81     |
+| Mathlib.Data.Finset.NoncommProd                                |        15409 |     913201 |    4.33     |
+| Mathlib.Algebra.BigOperators.Group.Finset                      |        78368 |     897792 |    4.38     |
+| Mathlib.Data.Finset.Preimage                                   |         4751 |     819423 |    4.62     |
 | Mathlib.Data.Set.Finite                                        |        36137 |     81467_ |    4.64     |
 | Mathlib.Data.Finite.Basic                                      |         3332 |     77853_ |    4.80     |
```

File: Mathlib/Analysis/Normed/Group/Lemmas.lean [hunk 1]
```diff
@@ -1,25 +1,291 @@
...
+
+In this file we define 10 classes:
 
-This file contains further lemmas about normed groups, requiring heavier imports than
-`Mathlib/Analysis/Normed/Group/Basic.lean`.
+* `Norm`, `NNNorm`: auxiliary classes endowing a type `α` with a function `norm : α → ℝ`
+  (notation: `‖x‖`) and `nnnorm : α → ℝ≥0` (notation: `‖x‖₊`), respectively;
+* `Seminormed...Group`: A seminormed (additive) (commutative) group is an (additive) (commutative)
+  group with a norm and a compatible pseudometric space structure:
+  `∀ x y, dist x y = ‖x / y‖` or `∀ x y, dist x y = ‖x - y‖`, depending on the group operation.
```
Comment (YaelDillies) [inline Mathlib/Analysis/Normed/Group/Lemmas.lean:24]: I think the former module doc was more accurate?

Comment (Ruben-VandeVelde) [reply Mathlib/Analysis/Normed/Group/Lemmas.lean:24]: Oops, I meant to fix that. Thanks for reminding me

**Comment by grunweg** [comment]
Thanks for splitting up this huge file! Would you also like to update the style-exceptions, or do you prefer e.g. me doing that once in a separate PR?

**Comment by Ruben-VandeVelde** [comment]
> Thanks for splitting up this huge file! Would you also like to update the style-exceptions, or do you prefer e.g. me doing that once in a separate PR?

Done

**Review by grunweg — COMMENTED** [review]
I see a point in moving these out of Normed.Group.Basic - yet I also have the impression that the `Lemmas` file is an incoherent bag of lemmas. You're not actively making this *worse*, though; you moved coherent chunks (as far as I can tell).
Is there some result in `Lemmas` which deserves mention in the module docstring? (Currently, I'd have to scroll through the entire file...) Feel free to maintainer merge with that addressed; this one deserves a second look.

I presume this PR is faithful code motion, but haven't verified this.

**Comment by Ruben-VandeVelde** [comment]
You inspired me to take another look and it turns out that performing some more coherent splits (along with Yaël's work) already improves the longest pole; see #14035. I'm not planning to continue work on this PR right now - if someone else wants to pick it up, go ahead.

**Review by YaelDillies — APPROVED** [review]
LGTM

maintainer merge

**Comment by YaelDillies** [comment]
Whoops sorry I wrote the previous comment in the wrong PR. I will close this PR for now. We can reopen it later need it be.

**Pull request closed without merging**